libcore/AbstractSlot.vala

```diff
@@ -68,17 +67,19 @@ namespace Files {
         public signal void directory_loaded (Files.Directory dir);
 
         public void add_extra_widget (Gtk.Widget widget) {
-            extra_location_widgets.pack_start (widget);
+            extra_location_widgets.prepend (widget);
         }
 
         public void add_extra_action_widget (Gtk.Widget widget) {
-            extra_action_widgets.pack_start (widget);
+            extra_action_widgets.prepend (widget);
         }
 
         public void add_overlay (Gtk.Widget widget) {
-            overlay = new Gtk.Overlay ();
-            content_box.pack_start (overlay, true, true, 0);
-            overlay.add (widget);
+            overlay = new Gtk.Overlay () {
+                child = widget
+            };
+            content_box.prepend (overlay);
+            overlay.child = widget;
```

Comment:
Child is set twice
```suggestion
```

Reply:
I wouldn't get too picky about things in AbstractSlot and several other classes which do not exist in `gtk4` anyway.  This gets replaced by a simpler interface.